src/main/java/org/checkerframework/specimin/SpeciminRunner.java

```diff
@@ -442,6 +435,65 @@ private static void performMinimizationImpl(
       cu.accept(mustImplementMethodsVisitor, null);
     }
 
+    Set<CompilationUnit> compilationUnitsToSolveAnnotations =
+        new HashSet<>(parsedTargetFiles.values());
+
+    // This should be safe to run after MustImplementMethodsVisitor because 
+    // annotations do not inherit
+    AnnotationParameterTypesVisitor annotationParameterTypesVisitor =
+        new AnnotationParameterTypesVisitor(
+            targetFieldNames,
+            finder.getTargetMethods(),
+            solveMethodOverridingVisitor.getUsedMembers(),
+            solveMethodOverridingVisitor.getUsedClass());
+
+    relatedClass = new HashSet<>(parsedTargetFiles.keySet());
+
+    while (!compilationUnitsToSolveAnnotations.isEmpty()) {
+      for (CompilationUnit cu : compilationUnitsToSolveAnnotations) {
+        cu.accept(annotationParameterTypesVisitor, null);
+      }
+
+      // add all files related to the target annotations
+      for (String annoFullName : annotationParameterTypesVisitor.getClassesToAdd()) {
+        String directoryOfFile = annoFullName.replace(".", "/") + ".java";
+        File thisFile = new File(root + directoryOfFile);
+        // classes from JDK are automatically on the classpath, so UnsolvedSymbolVisitor will not
+        // create synthetic files for them
```

Comment:
Sorry, the code used here was directly copy-pasted from above (line 370-ish), and I didn't update the comment. I removed the comment here (lines 461/462)since it isn't relevant to the code.